linux/swrandom/swrandom.c

```diff
@@ -2003,22 +2008,39 @@ static int acm_readdir(const char* path, acm_readdir_t filler, void* context)
 /**
  * A function used when searching for SwiftRNG ACM devices
  */
-static int acm_filldir_callback(void* data, const char *name, int nameLength, loff_t offset, u64 ino, unsigned int dType)
+static
+#ifdef TL_MIN_KERNEL_6_1
+       bool
+#else
+       int
+#endif
+            acm_filldir_callback(void* data, const char *name, int nameLength, loff_t offset, u64 ino, unsigned int dType)
 {
+#ifdef TL_MIN_KERNEL_6_1
+	   bool stop_ret = false;
+	   bool go_on_ret = true;
+#else
+	   int stop_ret = 1;
+	   int go_on_ret = 0;
```

Comment:
Shouldn't this be reversed to match the values in the TL_MIN_KERNEL_6_1 version above?

```suggestion
	   int stop_ret = 0;
	   int go_on_ret = 1;
```

Reply:
That is how the old API is treating int response: 0 - means all is good, go on iterating through the next directory item. The new API, however is using boolean true value for the same intension.